arduino/seed_aura/seed_aura.ino

```diff
@@ -0,0 +1,186 @@
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    mplsartindustry/daisy is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with mplsartindustry/daisy.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+#include "DaisyDuino.h"
+
+DaisyHardware seed;
+
+#define DEPTH_HZ 400.0f
+#define MIN_FREQ_HZ 200.0f
+#define MIN_LFO_RATE_HZ 0.001f
+#define MAX_LFO_RATE_HZ 400.0f
+
+MoogLadder lpf_l;
+MoogLadder lpf_r;
+Oscillator lfo_l;
+Oscillator lfo_r;
+
+/*
+
+  Daisy Seed pin mapping with MA&I pcb board
+
+  11  D10     --> SW1
+  16  in[0]   --> AUDIO_IN_L
+  17  in[1]   --> AUDIO_IN_R
+  18  out[0]  --> AUDIO_OUT_L
+  19  out[1]  --> AUDIO_OUT_R
+  23  A1/D16  --> POT1/J7 pin 2
+  24  A2/D17  --> POT2/J8 pin 2
+  25  A3/D18  --> POT3/J9 pin 2
+  29  A7/D22/DAC OUT 2 --> LEDs pin 1
+  30  A8/D23/DAC OUT 1 --> LEDs pin 3
+  31  A9/D24  --> FS1
+
+ */
+
+#define PIN_SW_1 D10
+#define PIN_POT_1 A1
+#define PIN_POT_2 A2
+#define PIN_POT_3 A3
+#define PIN_LED_1 A7
+#define PIN_LED_2 A8
+#define PIN_FS_1 D24
+
+void audio(float **in, float **out, size_t size) {
+
+  // digital read switch
+  int sw_1 = readSwitch(PIN_SW_1);
+
+  // analog read pots; pots are wired backward
+  uint32_t pot_1 = readPot(PIN_POT_1);
+  uint32_t pot_2 = readPot(PIN_POT_2);
+  uint32_t pot_3 = readPot(PIN_POT_3);
+
+  // analog write to led 1
+  writeLed(PIN_LED_1, pot_1 * 0.66f + pot_2 * 0.33f);
+
+  // map pot_1 to presence
+  float presence = mapf(pot_1, 0.0f, 1023.0f, 0.0f, 1.0f);
+  // map pot_2 to gain
+  float gain = mapf(pot_2, 0.0f, 1023.0f, 0.0f, 1.8f);
+
+  if (sw_1 == LOW) {
+
+    // map pot_3 to freq
+    float freq = mapf(pot_3, 0.0f, 1023.0f, 30.0f, 3000.0f);
+
+    // analog write to led 2
+    writeLed(PIN_LED_2, pot_3 * 0.66f + pot_2 * 0.33f);
+
+    // set filter frequency
+    lpf_l.SetFreq(freq);
+    lpf_r.SetFreq(freq);
+  }
+  else if (sw_1 == HIGH) {
+
+    // map pot_3 to lfo rate
+    float lfoRate = mapf(pot_3, 0.0f, 1023.0f, MIN_LFO_RATE_HZ, MAX_LFO_RATE_HZ);
+
+    // update lfo rate
+    lfo_l.SetFreq(lfoRate);
+    lfo_r.SetFreq(lfoRate);
+    float v_l = lfo_l.Process();
+    float v_r = lfo_r.Process();
+
+    // analog write to led 2
+    writeLed(PIN_LED_2, mapf(v_l, -1.0f * DEPTH_HZ, DEPTH_HZ, 0.0f, 1.0f) * 1023.0f);
+
+    // set filter frequency
+    lpf_l.SetFreq(MIN_FREQ_HZ + (DEPTH_HZ / 2.0f) + v_l);
+    lpf_r.SetFreq(MIN_FREQ_HZ + (DEPTH_HZ / 2.0f) + v_r);
+  }
+
+  // apply gain, presence, soft clip
+  for (size_t i = 0; i < size; i++) {
+
+    float pre_l = in[0][i];
```

Comment:
This is the part where perhaps I should be using `last` instead to implement feedback